ckanext/scheming/templates/scheming/form_snippets/datetime.html

```diff
@@ -0,0 +1,42 @@
+{% import 'macros/form.html' as form %}
+{% set date = '' %}
+{% set time = '' %}
+{% set dateTime = data.get(field.field_name, '') %}
+
+{% if dateTime is string %}
+    {% set parts = dateTime.split() %}
+    {% set date = parts[0] %}
+    {% set time = parts[1] %}
+{% elif dateTime is sequence %}
+    {% set date = dateTime[0] %}
+    {% set time = dateTime[1] %}
+{% endif %}
+{% call form.input(
+    field.field_name,
+    id='field-' + field.field_name +  '-date',
+    label= h.scheming_language_text(field.label),
+    type='date',
+    value=date,
+    error=errors[field.field_name],
+    classes=['control-medium'],
+    attrs=field.form_attrs if 'form_attrs' in field else {},
+    is_required=h.scheming_field_required(field)
+    )
+%}
+    {%- snippet 'scheming/form_snippets/help_text.html', field=field -%}
+{% endcall %}
+
+{% call form.input(
+    field.field_name,
+    id='field-' + field.field_name + '-time',
+    label= h.scheming_language_text(field.get('label_time', 'Time')),
+    type='time',
+    value=time,
+    error=errors[field.field_name],
```

Comment:
If you give the fields different names you can use the 4-argument version of the validator to associate errors with the separate fields. It will only work with the form, but I think that's what you want